src/main/scala/cromwell/engine/workflow/WorkflowOptions.scala

```
@@ -0,0 +1,89 @@
+      case Success(encryptedValue) => Success(JsObject(Map(
+        "iv" -> JsString(encryptedValue.base64Iv),
+        "ciphertext" -> JsString(encryptedValue.base64CipherText)
+      )))
+      case Failure(ex) => Failure(ex)
+    }
+  }
+
+  def decryptField(obj: JsObject): Try[String] = {
+    (obj.fields.get("iv"), obj.fields.get("ciphertext")) match {
+      case (Some(iv: JsString), Some(ciphertext: JsString)) =>
+        Aes256Cbc.decrypt(EncryptedBytes(ciphertext.value, iv.value), SecretKey(WorkflowOptions.EncryptionKey)) match {
+          case Success(decryptedBytes) => Success(new String(decryptedBytes, "utf-8"))
+          case Failure(ex) => Failure(ex)
+        }
+      case _ => Failure(new Throwable(s"JsObject must have 'iv' and 'ciphertext' fields to decrypt: $obj"))
+    }
+  }
+
+  def isEncryptedField(jsValue: JsValue): Boolean = jsValue match {
+    case obj: JsObject if obj.fields.keys.exists(_ == "iv") && obj.fields.keys.exists(_ == "ciphertext") => true
+    case _ => false
+  }
+
+  def fromJsonObject(jsObject: JsObject): Try[WorkflowOptions] = {
+    val encrypted = jsObject.fields map {
+      case (k, v: JsString) if EncryptedFields.contains(k) => k -> encryptField(v)
+      case (k, v: JsString) => k -> Success(v)
+      case (k, v) if isEncryptedField(v) => k -> Success(v)
+      case (k, v) => k -> Failure(new Throwable(s"Unsupported key/value pair in WorkflowOptions: $k -> $v"))
+    }
+
+    encrypted.values collect { case f: Failure[_] => f } match {
+      case s if s.nonEmpty => Failure(new Throwable(s.map(_.exception.getMessage).mkString("\n")))
+      case _ => Success(WorkflowOptions(new JsObject(encrypted map { case (k, v) => k -> v.get })))
+    }
+  }
+
+  def fromJsonString(json: String): Try[WorkflowOptions] = Try(json.parseJson) match {
+    case Success(obj: JsObject) => fromJsonObject(obj)
+    case _ => Failure(new Throwable("Expecting a JSON object"))
```

This has swallowed the actual error